### PR TITLE
[CPU][ARM] Fixes in SDPA kernels

### PR DESCRIPTION
**Pull request by alvoron**
### Details:
 - SVE/NEON exp fix: Clamp before range reduction, compute r with original n, then use n-1 only for scale; build scale via (n_int + 127) << 23 to match AVX.
 - ARM fp16 softmax: Use fp16 minimum instead of -FLT_MAX; handle attn_mask as f32 or f16 correctly; avoid bf16 dispatch on ARM64; fix dispatch order.
 - Test update: Increase sequence length from 10 to 16 to trigger SVE code

### Tickets:
 - CVS-178089

### CODE REVIEW

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp [hunk 1]
```diff
@@ -585,7 +586,16 @@ inline void scale_add2_reduce_max(ov::float16* a,
         }
 
         if (has_attn_mask) {
-            svfloat16_t v_mask = svld1_f16(pg_f16, reinterpret_cast<const float16_t*>(attn_mask + i));
+            svfloat16_t v_mask;
+            if constexpr (std::is_same_v<T, float>) {
+                svbool_t pg = svwhilelt_b32(i, i + vec_len_f16_neon);
```
[the remark this answers is not in the document]
Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp]: `vec_len_f16_neon` represents the number of elements in fp16 vector - 8.
By calling `svwhilelt_b32(i, i + 8)` a new predicate of 8 elements is created that will become one FP16 vector.

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/common.hpp
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/common.hpp:1]: Subtracting 1 from the n term in the reduced representation e^x = 2^n * e^r allows to center r around zero in the interval [-ln(2)/2; ln(2)/2], which leads to more numerically stable result. Multiplying then by 2 simply compensates this subtraction.
So, please revert this changes.

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp [hunk 2]
```diff
@@ -552,12 +552,13 @@ inline void scale_add2_reduce_max(ov::float16* a,
                                   float alibi_slope,
                                   ov::float16& max) {
     size_t i = 0;
+    float16_t min_f16 = std::numeric_limits<float16_t>::lowest();
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp]:
```suggestion
    constexpr float16_t min_f16 = std::numeric_limits<float16_t>::lowest();
```

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp [hunk 3]
```diff
@@ -585,7 +586,16 @@ inline void scale_add2_reduce_max(ov::float16* a,
...
-            svfloat16_t v_mask = svld1_f16(pg_f16, reinterpret_cast<const float16_t*>(attn_mask + i));
+            svfloat16_t v_mask;
+            if constexpr (std::is_same_v<T, float>) {
+                svbool_t pg_f32 = svwhilelt_b32(i, i + vec_len_f16_neon);
+                svfloat32_t m = svld1(pg_f32, attn_mask + i);
+                v_mask = svcvt_f16_f32_x(pg_f32, m);
+            } else if constexpr (std::is_same_v<T, ov::float16>) {
+                v_mask = svld1_f16(pg_f16, reinterpret_cast<const float16_t*>(attn_mask + i));
+            } else {
+                OPENVINO_THROW("attn_mask must be float or float16 type.");
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp]: Should be a static_assert?

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp [hunk 4]
```diff
@@ -585,7 +586,16 @@ inline void scale_add2_reduce_max(ov::float16* a,
         }
 
         if (has_attn_mask) {
-            svfloat16_t v_mask = svld1_f16(pg_f16, reinterpret_cast<const float16_t*>(attn_mask + i));
+            svfloat16_t v_mask;
+            if constexpr (std::is_same_v<T, float>) {
+                svbool_t pg_f32 = svwhilelt_b32(i, i + vec_len_f16_neon);
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp]: This is SVE code. Shouldn't we use SVE specific constants like `vec_len_f16_sve` and `vec_len_f32_sve` ?
Also, in the case of f32 attention_mask, it looks like we have to convert f32 in two steps (just like it's done for neon) as the fp16 SIMD vector capacity 2x of fp32 vector capacity and narrowing conversion happens leaving the upper word in each double word undefined.

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp (hunk 2 above)
Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp]: Added this suggestion.

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp (hunk 3 above)
Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp]: Replaced by static_assert

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp (hunk 4 above)
Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp]: Updated the fix

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/common.hpp
Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/common.hpp:1]: Reverted this change

File: src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp [hunk 5]
```diff
@@ -1378,31 +1411,33 @@ inline void attn_softmax_kernel<ov::float16>(ov::float16* a,
                                                     scale_add2_reduce_max<true, true, true>};
     int dispatch = (alibi ? 0b100 : 0) | (attn_mask ? 0b010 : 0) | (causal_mask ? 0b001 : 0);
     ov::float16 max = std::numeric_limits<ov::float16>::lowest();
-    if (attn_mask_prec == ov::element::f32) {
-        funcs_fp32[dispatch](a,
+    if (attn_mask_prec == ov::element::f16) {
```
Comment (alvoron) [inline src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp:1414]: I changed the order of precisions to get `fp32` case as a fallback (else branch).
